rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp

```diff
@@ -138,7 +138,24 @@ namespace rsx
 				glPixelStorei(GL_UNPACK_SWAP_BYTES, GL_TRUE);
 				glPixelStorei(GL_UNPACK_ROW_LENGTH, pitch / 2);
 
-				// TODO: texture swizzling
+				if (is_swizzled)
+				{
+					u16 *src, *dst;
+					u16 height = tex.height();
+					u16 width = tex.width();
+
+					unswizzledPixels = (u8*)malloc(width * height * 2);
```

Comment:
We tend to avoid juggling with memory directly, from what I can tell. Wouldn't it be safer/better to simply to use std::unique_ptr here? But Nekoteki probably can correct me on that.